src/DatePicker.js

```diff
@@ -13,8 +12,13 @@ class DatePicker extends React.Component {
 
   componentDidMount() {
     if (typeof M !== 'undefined') {
+      const { onChange } = this.props;
       const elem = document.getElementById(this.id);
-      this.instance = M.Datepicker.init(elem, this.props.options);
+      const options = onChange
+        ? { ...this.props.options, onSelect: onChange }
+        : this.props.options;
+
+      this.instance = M.Datepicker.init(elem, options);
```

Comment:
Nice! this could also be done like so, might be cleaner:

```jsx
const { onChange: () => {}, ...other } = this.props;
const elem = document.getElementById(this.id);

this.instance = M.Datepicker.init(elem, onSelect={onChange} {...other});
```

Is this cleaner or harder to understand?

Reply:
I think it's a little bit harder to grasp, what do you think?